pilot/pkg/leaderelection/leaderelection.go

```diff
@@ -177,10 +178,12 @@ func (l *LeaderElection) create() (*k8sleaderelection.LeaderElector, error) {
 		// to instances are both considered the leaders. As such, if this is intended to be use for mission-critical
 		// usages (rather than avoiding duplication of work), this may need to be re-evaluated.
 		ReleaseOnCancel: true,
+	}
+	if !l.perRevision {
```

Comment:
not clear to me, how is the original issue related to revision, can you clarify

Reply:
The key mechanism puts a "holderKey" in the lease. Then we compare our key to the lease.

The problem is with Lease type (i.e. when l.perRevision is set), there is no `holderKey` value, so its always `""`. This makes it constantly lose and flipflop between leaders